core/src/main/java/io/temporal/samples/hello/HelloUpdate.java

```diff
@@ -0,0 +1,281 @@
+    WorkflowClient.start(workflow::getGreetings);
+
+    // After start for getGreeting returns, the workflow is guaranteed to be started.
+    // So we can send an update to it using the workflow stub.
+    // This workflow keeps receiving updates until exit is called
+
+    // When the workflow is started the getGreetings will block for the previously defined
+    // conditions
+    // Send the first workflow update
+    workflow.waitForName("World");
+
+    /*
+     * Here we create a new workflow stub using the same workflow id.
+     * We do this to demonstrate that to send an update to an already running workflow
+     * you only need to know its workflow id.
+     */
+    GreetingWorkflow workflowById = client.newWorkflowStub(GreetingWorkflow.class, WORKFLOW_ID);
+
+    // Send the second update to our workflow
+    workflowById.waitForName("Universe");
+
+    /*
+     * Create an untyped workflow stub to demonstrate sending an update
+     * with the untyped stub.
+     */
+    WorkflowStub greetingStub = client.newUntypedWorkflowStub(WORKFLOW_ID);
+    greetingStub.update("waitForName", Integer.class, "Temporal");
+
+    try {
+      // The update request will fail on a empty name and the exception will be thrown here.
+      workflowById.waitForName("");
+      System.exit(-1);
+    } catch (WorkflowUpdateException e) {
+      Throwable cause = Throwables.getRootCause(e);
+      /*
+       * Here we should get our originally thrown ApplicationError
+       * and the message "Cannot greet someone with an empty name".
+       */
+      System.out.println("\n Update failed, root cause: " + cause.getMessage());
+    }
+    // Send our update validators limit of 10 updates
+    Integer sentUpdates = workflowById.waitForName("Update");
+    while (sentUpdates < 10) {
+      sentUpdates = workflowById.waitForName("Again");
+    }
+
+    // The update request will be rejected because our validator will fail
+    try {
+      workflowById.waitForName("Will be rejected");
+      System.exit(-1);
+    } catch (WorkflowUpdateException e) {
+      Throwable cause = Throwables.getRootCause(e);
+      System.out.println("\n Update rejected: " + cause.getMessage());
+    }
+
+    // Now let's send our exit signal to the workflow
+    workflowById.exit();
+
+    /*
+     * We now call our getGreetings workflow method synchronously after our workflow has started.
+     * This reconnects our workflowById workflow stub to the existing workflow and blocks until
+     * a result is available. Note that this behavior assumes that WorkflowOptions are not configured
+     * with WorkflowIdReusePolicy.AllowDuplicate. If they were, this call would fail with the
+     * WorkflowExecutionAlreadyStartedException exception.
+     */
+    List<String> greetings = workflowById.getGreetings();
+
+    // Print our two greetings which were sent by signals
+    System.out.println(greetings);
+    System.exit(0);
```

Comment:
```suggestion
```

Reply:
I had this since all the other samples I saw had it, not sure why TBH. Don't think we run any `daemon` threads